### PR TITLE
DONT MERGE: fix: change install path for atomic desktop support

### PR DESCRIPTION
**Pull request by boredsquirrel**
https://github.com/containers/ramalama/issues/177

I am still getting errors and would like to debug them

```fish
~/.local/bin ❯❯❯ cat patched-install.py | sudo python3
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100   883  100   883    0     0  13763      0 --:--:-- --:--:-- --:--:-- 13796
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100 13994  100 13994    0     0  47045      0 --:--:-- --:--:-- --:--:-- 46959
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100  2338  100  2338    0     0   8050      0 --:--:-- --:--:-- --:--:--  8062
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100  3538  100  3538    0     0  12891      0 --:--:-- --:--:-- --:--:-- 12912
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100  3676  100  3676    0     0  14003      0 --:--:-- --:--:-- --:--:-- 13977
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100  2169  100  2169    0     0   8514      0 --:--:-- --:--:-- --:--:--  8539
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100   218  100   218    0     0    751      0 --:--:-- --:--:-- --:--:--   754
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100  4038  100  4038    0     0   9777      0 --:--:-- --:--:-- --:--:--  9777
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100   118  100   118    0     0    422      0 --:--:-- --:--:-- --:--:--   422
~/.local/bin ❯❯❯ ramalama
Traceback (most recent call last):
  File "/usr/local/bin/ramalama", line 36, in <module>
    main(sys.argv[1:])
  File "/usr/local/bin/ramalama", line 17, in main
    import ramalama
```

### CODE REVIEW

File: install.py [hunk 1]
```diff
@@ -85,7 +85,7 @@ def setup_ramalama(bindir, tmp_dir):
     to_file = os.path.join(tmp_dir, from_file)
     download(url, to_file)
     ramalama_bin = os.path.join(bindir, binfile)
-    syspath = "/usr/share/ramalama"
+    syspath = "/usr/local/share/ramalama"
     if sys.platform == "darwin":
         install_mac_dependencies()
```
Comment (boredsquirrel) [inline install.py:90]: maybe we need another function here, to detect atomic desktops and just use a different path here?

**Comment by ericcurtin** [comment]
I would recommend grepping for:

```
/usr/share/ramalama
```

and everywhere that exists doing something like:

```
    sharedirs = ["/opt/homebrew/share/ramalama", "/usr/local/share/ramalama", "/usr/share/ramalama"]
    syspath = next((d for d in sharedirs if os.path.exists(d)), None)
```

File: install.py (hunk 1 above)
Comment (ericcurtin) [reply install.py:90]: Nice granite-code usage :)

I would recommend this fix:

```
    sharedirs = ["/opt/homebrew/share", "/usr/local/share", "/usr/share/ramalama"]
    syspath = next((d for d in sharedirs if os.path.exists(d)), None)
    syspath += "/ramalama"
    if sys.platform == "darwin":
        install_mac_dependencies()
```

for this part of the code, but I think this has to be fixed in multiple places.

But, if we use the logic, use "/opt/homebrew/share" if it exists, use "/usr/local/share" if it exists and then use "/usr/share/ramalama" if it exists. I think it should work everywhere.

After we have added the "/ramalama" dir in this script, everywhere else it's better to check for existence of:

```
    sharedirs = ["/opt/homebrew/share/ramalama", "/usr/local/share/ramalama", "/usr/share/ramalama/ramalama"]
```

**Comment by boredsquirrel** [comment]
so in this case, we would test for immutability, place the executable and then make all the rest take its information from the location of that executable?

**Comment by ericcurtin** [comment]
If you want to add a test for immutability, why not I guess... But checking for:

"/usr/local/share"

before

"/usr/share"

is simpler and should work just as well

**Comment by ericcurtin** [comment]
It'll fail anyway if /usr/share/local is not writable...

**Comment by ericcurtin** [comment]
Also, it's arguably more correct to use /usr/local over /usr in the curl | sh script... /usr is more for rpms

**Comment by ericcurtin** [comment]
We want to prefer:

/usr/share/local 

over

/usr/share

in macOS x86_64 also, /usr/share isn't really writable in macOS...

**Comment by boredsquirrel** [comment]
so my initial thought was correct, simply using `/usr/local/share` (i.e. `/var/usrlocal/share` on Fedora Atomic, doesnt matter) is a better way for the general Unix-like desktop?

Then I am fine with not testing for immutability and placing it there immediately.

**Pull request closed without merging**